platform-sdk/swirlds-common/src/main/java/com/swirlds/common/formatting/UnitFormatter.java

```diff
@@ -160,87 +158,177 @@ public UnitFormatter setShowSpaceInBetween(final boolean showSpaceInBetween) {
-                }
-            }
+    /**
+     * Add a term to the string builder, and return the remainder
+     *
+     * @param stringBuilder    the string builder to write the rendered string to
+     * @param startingQuantity the starting quantity
+     * @return the remainder
+     */
+    @NonNull
+    private Unit.SimplifiedQuantity<?> addTerm(
+            @NonNull final StringBuilder stringBuilder, @NonNull final Unit.SimplifiedQuantity<?> startingQuantity) {
+
+        final long roundedDownQuantity = (long) startingQuantity.quantity();
+
+        final Unit.SimplifiedQuantity<?> remainder = new Unit.SimplifiedQuantity<>(
+                startingQuantity.quantity() - roundedDownQuantity, startingQuantity.unit());
+        final Unit.SimplifiedQuantity<?> simplifiedRemainder = remainder.unit().simplify(remainder.quantity());
+
+        final String termString;
+        if (simplifiedRemainder.unit().ordinal() < unit.ordinal()) {
+            termString = commaSeparatedNumber(startingQuantity.quantity(), decimalPlaces);
+        } else {
+            termString = commaSeparatedNumber(roundedDownQuantity, 0);
+        }
+
+        stringBuilder.append(termString);
+
+        // multi simplified format always displays the unit
+        appendUnit(stringBuilder, termString, startingQuantity.unit());
+
+        return simplifiedRemainder;
+    }
+
+    /**
+     * Format the number with multiple units.
+     *
+     * @param stringBuilder the string builder to write the rendered string to
+     */
+    private void formatWithMultiSimplifiedUnits(@NonNull final StringBuilder stringBuilder) {
+        double originalQuantity = longQuantity == null ? doubleQuantity : (double) longQuantity;
+
+        Unit.SimplifiedQuantity<?> remainder = addTerm(
+                stringBuilder,
+                new Unit.SimplifiedQuantity<>(originalQuantity, unit).unit().simplify(originalQuantity));
+
+        while (remainder.unit().ordinal() >= unit.ordinal()) {
+            stringBuilder.append(" ");
+            remainder = addTerm(stringBuilder, remainder);
+        }
+    }
+
+    /**
+     * Render the number and write it to a string builder
+     *
+     * @param stringBuilder the string builder to write the rendered string to
+     */
+    public void render(@NonNull final StringBuilder stringBuilder) {
+        Objects.requireNonNull(stringBuilder);
+
+        if (longQuantity == null && doubleQuantity == null) {
+            throw new IllegalStateException("Quantity must be set");
+        }
+
+        if (longQuantity != null && doubleQuantity != null) {
+            throw new IllegalStateException("Quantity must be set to either long or double, not both");
+        }
+
+        switch (unitFormat) {
+            case UNSIMPLIFIED -> formatWithUnsimplifiedUnit(stringBuilder);
+            case SINGLE_SIMPLIFIED -> formatWithSingleSimplifiedUnit(stringBuilder);
+            case MULTI_SIMPLIFIED -> formatWithMultiSimplifiedUnits(stringBuilder);
```

Comment:
Would it make sense to have an optional setting that allows the caller to specify the number of units to show? For example, if we are writing a time that uses years and months, we probably don't need to report nanoseconds. A good default seems like to use 3 units of precision, e.g. `3 months 4 days 7.2 hours` (note how the last term in the sequence uses decimal places)

Reply:
e.g.
```
 DataUnit.UNIT_KILOBYTES
                        .buildFormatter(XXXX)
                        .setUnitFormat(UnitFormat.MULTI_SIMPLIFIED)
                        .setMultiUnitCount(3) // ignored if unit format is not MULTI_SIMPLIFIED
                        .render());
```

Reply:
Lowest unit is ok for now. 

In the future, we may eventually want to also support the ability to toggle the number of units in case the caller doesn't know the approximate size of the quantity before hand (i.e. if we are unsure if a file is measured in bytes or terabytes).

Reply:
I just went ahead and implemented both, since it wasn't much work